### PR TITLE
fix hpdnn batchnorm decomp

### PR DESCRIPTION
**Pull request by dnikolaev-amd**
Decomposition for hipdnn_batch_norm
Fixes `test_decomp.py::HasDecompTest::test_has_decomposition`
Based on https://github.com/pytorch/pytorch/pull/125069

### CODE REVIEW

**Review by zjgarvey — APPROVED** [review]
Ah, I see. I didn't realize the automatic forward differentiation path / jvp stuff needs this. Yeah, LGTM

**Review by alugorey — APPROVED** [review]
lgtm
